src/main/java/org/apache/sysds/runtime/controlprogram/caching/CacheableData.java

```diff
@@ -229,7 +229,7 @@ protected CacheableData(DataType dt, ValueType vt) {
 		_uniqueID = _seq.getNextID();
 		_cacheStatus = CacheStatus.EMPTY;
 		_numReadThreads = 0;
-		_gpuObjects = DMLScript.USE_ACCELERATOR ? new HashMap<>() : null;
+		_gpuObjects = DMLScript.USE_ACCELERATOR ? new ConcurrentHashMap<>() : null;
```

Comment:
Is there a concrete reason for using ConcurrentHashMap?

Reply:
When useing multi-GPU, the size of `_gpuObjects` is large than 2. Then the `ConcurrentModificationError` will be occered. Using `ConcurrentHashMap` could avoid this. Of corse there still some other ways to deal with it. But I think it's the best solusion to avoid copying an extra GPU memory.